clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/caching/agents/TitusStreamingUpdateAgent.java

```diff
@@ -322,6 +328,18 @@ private void updateTask(StreamingCacheState state, Task task) {
       state.changes.incrementAndGet();
     }
 
+    private void updateMovedTask(StreamingCacheState state, Task task, String destinationJobId,
+                                 String sourceJobId) {
+      if (sourceJobId != null) {
+        if (state.tasks.containsKey(sourceJobId)) {
+          state.tasks.get(destinationJobId).add(task);
```

Comment:
If the destination job doesn't already have running tasks,`state.tasks.get(destinationJobId)` will return null and the add would throw an NPE. `state.tasks.computeIfAbsent(jobId, t -> new HashSet<>()).add(task)` or something similar would be safer.